install/0000_90_cluster-version-operator_02_servicemonitor.yaml

```diff
@@ -46,6 +46,13 @@ spec:
         severity: critical
   - name: cluster-operators
     rules:
+    - alert: ClusterNotUpgradeable
+      annotations:
+        message: One or more cluster operators have been blocking minor version cluster upgrades for at least an hour for reason {{ "{{ with $cluster_operator_conditions := \"cluster_operator_conditions\" | query}}{{range $value := .}}{{if and (eq (label \"name\" $value) \"version\") (eq (label \"condition\" $value) \"Upgradeable\") (eq (label \"endpoint\" $value) \"metrics\") (eq (value $value) 0.0) (ne (len (label \"reason\" $value)) 0) }} {{label \"reason\" $value}} {{end}}{{end}}{{end}}" }}. {{ "{{ with $console_url := \"console_url\" | query }}{{ if ne (len (label \"url\" (first $console_url ) ) ) 0}} For more information refer to {{ label \"url\" (first $console_url ) }}/settings/cluster/.{{ end }}{{ end }}" }}
+      expr: |
+        cluster_operator_conditions{name="version",condition="Upgradeable",endpoint="metrics"} == 0
+      labels:
+        severity: critical
```

Comment:
Not quite sure its worth "critical" alert. Enabling unsupported features is intended and makes cluster un-upgradable, but it doesn't mean it requires immediate attention.

Perhaps it should be "warning"?

Reply:
Yeah, seems reasonable and the existing TechPreviewNoUpgrade is a Warning.